src/algebra/big_operators/order.lean

```diff
@@ -231,21 +232,50 @@ end
 
 end linear_ordered_cancel_comm_monoid
 
+section linear_ordered_comm_monoid_with_zero
+variables [linear_ordered_comm_monoid_with_zero β]
+
+lemma prod_nonneg' {s : finset α} {f : α → β}
+  (h0 : ∀(x ∈ s), 0 ≤ f x) : 0 ≤ (∏ x in s, f x) :=
+begin
```

Comment:
Does `linear_ordered_comm_ring` not imply `linear_ordered_comm_monoid_with_zero`? If it does, can't this proof replace `prod_nonneg`?

Reply:
No, `linear_ordered_comm_monoid_with_zero` is for things line `nnreal` where multiplication by *any fixed* element is a monotonically increasing function.